src/Symfony/Component/HttpClient/Response/CurlResponse.php

```diff
@@ -354,7 +354,7 @@ private static function select(ClientState $multi, float $timeout): int
             }
         }
 
-        if (0 !== $selected = curl_multi_select($multi->handle, $timeout)) {
+        if (0 !== $selected = curl_multi_select($multi->handle, $timeout < 0 ? $timeout : 1.1)) {
```

Comment:
Maybe `1.0` to match default from php. Or is there a specific reason for 1.1 ?
https://www.php.net/manual/de/function.curl-multi-select.php

Reply:
$timeout >= 0 ? $timeout : 1.0

Reply:
should then be like this I guess:

```suggestion
        if (0 !== $selected = curl_multi_select($multi->handle, 0 < $timeout ? $timeout : 1.0)) {
```

Reply:
I obviously need better glasses as I could swear when I checked the documentation it was 1.1 !! Ffs I suck. Sorry

Reply:
the important part though is to change the condition as with the previous code `$timeout` would have only been forwarded if it was less than `0` while it should probably be the other way around :)

Reply:
I see that now wow - what the hell was I thinking yesterday?!!?!